engine/script/src/script_http_js.cpp

```diff
@@ -53,12 +53,24 @@ namespace dmScript
         int             m_Callback;
     };
 
+    struct RequestParams
+    {
+        const char* m_Method;
+        const char* m_Url;
+        const char* m_Headers;
+        void* m_Args;
+        void* m_SendData;
+        size_t m_SendDataLength;
+        uint64_t m_RequestTimeout;
+    };
+
     extern "C" void dmScriptHttpRequestAsync(const char* method, const char* url, const char* headers, void* arg, OnLoad onload, OnError onerror, const void* send_data, int send_data_length, int timeout);
 
-    void HttpRequestAsync(const char* method, const char* url, const char* headers, void *arg, OnLoad ol, OnError oe, const void* send_data, int send_data_length)
+    void HttpRequestAsync(const RequestParams& request_params, OnLoad ol, OnError oe)
     {
-        int timeout  = (int) g_Timeout;
-        dmScriptHttpRequestAsync(method, url, headers, arg, ol, oe, send_data, send_data_length, timeout);
+        dmScriptHttpRequestAsync(request_params.m_Method, request_params.m_Url, request_params.m_Headers,
+            request_params.m_Args, ol, oe, request_params.m_SendData, request_params.m_SendDataLength, request_params.m_RequestTimeout);
+        free((void*) request_params.m_SendData);
```

Comment:
Free memory that was allocated here https://github.com/defold/defold/pull/8371/files#diff-8df27bb2adf2e376b26a55b64dd6d84fbec204dc69af00239bb14b3c6d3182d2R167

Reply:
If not a problem, I think we should release the memory on the caller site, after the HttpRequestAsync call.
That way the ownership/lifetime is clearer for the allocated memory.

Reply:
We don't store `request_params` after making `dmScriptHttpRequestAsync` call.
So do you want to store at least `request_params.m_SendData` in `RequestContext` struct and free it inside `SendResponse`?

Reply:
@JCash Please, confirm am I understand purpose correctly.

Reply:
What I meant was that, if it's ok to free the memory here, then the dmScriptHttpRequestAsync() has already copied the data safely?
And then it makes less sense to free the memory here, as opposed to outside of this function, at the caller site.

Reply:
Ahhh... Sorry.
When I tested changes I also thought about it. At js side it has direct access to the memory via `HEAPU8.subarray(send_data, send_data + send_data_length)`. But it passed to `XMLHttpRequest.send`. I'm npt sure that XMLHttpRequest take the ownership of the data. I'll look for some documentation regarding that.
At least during my test there were no issues with memory access. I'll recheck.

Reply:
I cannot find any information confirms that javascript part take ownership of memory block. So I've added void* to `RequestContext` structure and free allocated for `m_SendData` memory in request callback.